pkg/hcl_interpreter/term.go

```diff
@@ -265,39 +265,32 @@ func (t Term) Evaluate(
 	evalExpr func(expr hcl.Expression, extraVars cty.Value) (cty.Value, hcl.Diagnostics),
 ) (cty.Value, hcl.Diagnostics) {
 	if t.count != nil {
-		// Helper
-		parseCount := func(val cty.Value) *int64 {
-			if val.IsNull() || !val.IsKnown() {
```

Comment:
I don't think I understand how this helps fixing the bug. 

As far as I can tell, the new code can be considered a refactor for all cases except one:
```
!val.IsNull() && val.IsKnown() && !val.IsNumber()
```

(all other cases evaluate to the same values in both versions, if I'm correct?)

However, that case does not seem to be what you're describing as the bug, so I wonder about this change :) 
Is this part purely a refactor so that we use `count = 1` more consistently?

Reply:
You are right -- the actual fix for the bug would be just this (applied to `main` as of writing):

```diff
diff --git a/pkg/hcl_interpreter/term.go b/pkg/hcl_interpreter/term.go
index 2c5e1fe..46c3319 100644
--- a/pkg/hcl_interpreter/term.go
+++ b/pkg/hcl_interpreter/term.go
@@ -280,7 +280,8 @@ func (t Term) Evaluate(
 					return &count
 				}
 			}
-			return nil
+			count := int64(1)
+			return &count
 		}
 
 		diagnostics := hcl.Diagnostics{}
```

However, that makes the fact that we're using a pointer for `count` unnecessary, and at that point we may as well remove the `parseCount` helper entirely.